Assets/Plugins/iOS/Shopify/BuyTests/StringTests.swift

```diff
@@ -29,37 +29,60 @@ import PassKit
 
 @testable import ProductName
 
+@available(iOS 10.0, *)
 class StringTests : XCTestCase {
 
     /// Test that all rawValues are unique
     func testPaymentAuthorizationStatusUnique() {
         
-        let rawValues: [String] = PKPaymentAuthorizationStatus.allItems.map {
-            $0.rawValue
-        }
+        let states: [String] = [
+            PKPaymentAuthorizationStatus.failure.rawValue,
+            PKPaymentAuthorizationStatus.success.rawValue,
+            PKPaymentAuthorizationStatus.invalidBillingPostalAddress.rawValue,
+            PKPaymentAuthorizationStatus.invalidShippingPostalAddress.rawValue,
+            PKPaymentAuthorizationStatus.invalidShippingContact.rawValue,
+            PKPaymentAuthorizationStatus.pinRequired.rawValue,
+            PKPaymentAuthorizationStatus.pinLockout.rawValue,
+            PKPaymentAuthorizationStatus.pinIncorrect.rawValue]
         
-        XCTAssertEqual(rawValues.count, Set(rawValues).count)
+        XCTAssertEqual(states.count, Set(states).count)
     }
     
     func testPaymentSummaryItemTypeUnique() {
-        
-        let rawValues: [String] = PKPaymentSummaryItemType.allItems.map {
-            $0.rawValue
-        }
-        
-        XCTAssertEqual(rawValues.count, Set(rawValues).count)
+        let types: [String] = [PKPaymentSummaryItemType.pending.rawValue, PKPaymentSummaryItemType.final.rawValue]
+        XCTAssertEqual(types.count, Set(types).count)
     }
     
     /// Test that rawValues correspond to the init
     func testPaymentAuthorizationStatusInit() {
-        for status in PKPaymentAuthorizationStatus.allItems {
-            XCTAssertEqual(status, PKPaymentAuthorizationStatus(rawValue: status.rawValue as String))
+        
+        let assertInitIdempotency = { (status: PKPaymentAuthorizationStatus) in
+            XCTAssertEqual(
```

Comment:
Tests that that the 'rawValue' of an state, is the same value used for initialization of that state.

This fails when a 'rawValue' of a state is different than the raw value that would be given to the initialization to create that state.